### PR TITLE
cicd: examples arm64

### PR DESCRIPTION
**Pull request by maartenarnst**
The complex division example needed to be reworked because on ARM64, `FP_ILOGBNAN` maps to `INT_MAX`, making the previous implementation impractical.

With `logb`, we have:
- `+-0` -> `logb` -> `-\infty`
- `+-\infty` -> `logb` -> `+\infty`
- `nan` -> `logb` -> `nan`

With `ilogb`, we have:
- `+-0` -> `ilogb` -> `fp_ilogb0`
- `+-\infty` -> `ilogb` -> `int_max`
- `nan` -> `ilogb` -> `fp_ilogbnan` -> `int_min` on x86 vs `int_max` on arm64

So on arm64, we can't distinguish `+-\infty` from `nan` input.

The conditional that checks whether scaling will be used is ok, we exclude all infinities/nans.

In the conditional that ensures the compliance, the third branch is an issue. It wants to protect against the denominator being infinity. And so we must know whether we are in the `+-\infty` or `nan` case. The fix is to check based directly on the ilogb input, instead of based on the ilogb result.

### CODE REVIEW

File: docs/source/examples/kokkos/complex/division.rst [hunk 1]
```diff
@@ -11,17 +11,53 @@ computes :math:`c^2 + d^2`, which can overflow for large inputs. Underflow regim
...
-which mandates specific handling of edge cases such as infinite or zero operands.
-The :download:`compliance test <../../../../../examples/kokkos/complex/example_division_compliance.cpp>`
-can be used to validate several implementations against these requirements.
-Note that libraries such as `CCCL <https://github.com/NVIDIA/cccl/blob/a91db6e2a022a7aa03b37873f0d4caf5ac81281d/libcudacxx/include/cuda/std/__complex/complex.h#L469>`_
-expose an opt-out mechanism for this handling, which can save computation cycles when such operands are not expected.
+The standard **ISO/IEC 60559** :cite:`iso-iec-9899-2024` raises the issue of overflow
+and underflow and specifies the expected outcomes for the edge cases.
+It also suggests an implementation that adopts a scaling approach using the :code:`scalbn` function
+and relies on branching to handle the edge cases.
+Compilers such as GCC and LLVM and libraries such as `CCCL <https://github.com/NVIDIA/cccl/blob/a91db6e2a022a7aa03b37873f0d4caf5ac81281d/libcudacxx/include/cuda/std/__complex/complex.h#L469>`_
```
Comment (romintomasetti) [inline docs/source/examples/kokkos/complex/division.rst]: Let's add the permalinks.

Comment (maartenarnst) [reply docs/source/examples/kokkos/complex/division.rst]: Okay 📚!

**Review by romintomasetti — APPROVED** [review]

File: docs/source/examples/kokkos/complex/division.rst [hunk 2]
```diff
@@ -11,17 +11,54 @@ computes :math:`c^2 + d^2`, which can overflow for large inputs. Underflow regim
...
-Some implementations also claim **ISO/IEC 60559** compliance :cite:`iso-iec-9899-2024`,
-which mandates specific handling of edge cases such as infinite or zero operands.
-The :download:`compliance test <../../../../../examples/kokkos/complex/example_division_compliance.cpp>`
-can be used to validate several implementations against these requirements.
-Note that libraries such as `CCCL <https://github.com/NVIDIA/cccl/blob/a91db6e2a022a7aa03b37873f0d4caf5ac81281d/libcudacxx/include/cuda/std/__complex/complex.h#L469>`_
-expose an opt-out mechanism for this handling, which can save computation cycles when such operands are not expected.
+The standard **ISO/IEC 60559** :cite:`iso-iec-9899-2024` raises the issue of overflow
+and underflow and specifies the expected outcomes for the edge cases.
+It also suggests an implementation that adopts a scaling approach using the :code:`scalbn` function
+and relies on branching to handle the edge cases.
```
[the remark this answers is not in the document]
Comment (maartenarnst) [reply docs/source/examples/kokkos/complex/division.rst]: OK. Changed indeed.

**Review by romintomasetti — APPROVED** [review]
🖇️
